packages/developer-extension/src/background/domain/flushEvents.ts

```diff
@@ -0,0 +1,13 @@
+import { listenAction } from '../actions'
+import { evaluateCodeInActiveTab } from '../utils'
+
+listenAction('flushEvents', () =>
+  // Simulates a brief page visibility change (visible > hide > visible)
+  evaluateCodeInActiveTab(`
+    const descriptor = Object.getOwnPropertyDescriptor(Document.prototype, 'visibilityState')
+    Object.defineProperty(Document.prototype, 'visibilityState', { value: 'hidden' })
+    document.dispatchEvent(new Event('visibilitychange', { bubbles: true }))
+    Object.defineProperty(Document.prototype, 'visibilityState', descriptor)
+    document.dispatchEvent(new Event('visibilitychange', { bubbles: true }))
```

Comment:
I would prefer to not write code as string, what about the following
```ts
const flushEvents = () => {
    const descriptor = Object.getOwnPropertyDescriptor(Document.prototype, 'visibilityState')
    Object.defineProperty(Document.prototype, 'visibilityState', { value: 'hidden' })
    document.dispatchEvent(new Event('visibilitychange', { bubbles: true }))
    Object.defineProperty(Document.prototype, 'visibilityState', descriptor)
    document.dispatchEvent(new Event('visibilitychange', { bubbles: true }))
}
evaluateCodeInActiveTab(`;(${flushEvents})()`)
```

Reply:
or even:
```ts
evaluateCodeInActiveTab(flushEvents)
```
and let the evaluate function deal with the stringification

Reply:
Ok. I tend to avoid such pattern as it is not obvious that they will be used as a string, but I'll make the change anyway